### ISSUE TITLE
Publish helm charts

### ISSUE THREAD

**Issue by dblock** [opensearch-project/opensearch-devops#12]
Coming from #10 and #4

**Comment by TheAlgo** [comment]
I can take this up once OpenSearch Dashboards Chart is merged 😄

**Comment by DandyDeveloper** [comment]
@TheAlgo If you need help let me know. I have a good example of GitHub Actions that do this. 

https://github.com/DandyDeveloper/charts/blob/master/.github/workflows/publish.yml

There's a lot of different options for doing this nowadays through GitHub actions. Let me know if I can help.

**Comment by TheAlgo** [comment]
@DandyDeveloper  yes that will be great we can pick it up together after the dashboards chart is merged.  We can decide whether we want to use actions or something else.

**Comment by smlx** [comment]
The chart version should probably be something less than `1.0.0` for the first release. Maybe `0.1.0`?

**Comment by davidkarlsen** [comment]
Existing examples of using action-based tooling for CI and publishing:
* https://github.com/evryfs/helm-charts
* https://github.com/helm/charts-repo-actions-demo/

**Comment by mprimeaux** [comment]
I have completed the GitHub workflow using the helm-chart-tester and helm-chart-releaser actions and am happy to contribute to this project is folks are supportive.

**Comment by mprimeaux** [comment]
```shell
helm repo add opensearch https://mprimeaux.github.io/opensearch-devops/
helm search repo open search
```

I'll re-fork and cleanup but just wanted solicit feedback.

**Comment by TheAlgo** [comment]
>I have completed the GitHub workflow using the helm-chart-tester and helm-chart-releaser actions and am happy to contribute to this project if folks are supportive.

I would be happy love to contribute and support. Thanks

**Comment by mprimeaux** [comment]
Thanks for your feedback. I'll re-fork, clean the patch up, and submit a PR. All comments and criticisms are always welcome. it's a learning opportunity for me and all.

**Comment by peterzhuamazon** [comment]
@mprimeaux Thanks for the work we are also in the process of moving helm to its own repo instead of keeping it in opensearch-devops repo.

Once it has its own repo could you apply the same changes you have in your fork to here?

Thanks.

**Comment by mprimeaux** [comment]
@peterzhuamazon Definitely. Just let me know when you're ready for the new fork and PR

**Comment by mprimeaux** [comment]
@peterzhuamazon What is the estimated time for when the new repository will be ready? Again, thanks for your help.

**Comment by peterzhuamazon** [comment]
Hi @mprimeaux we need to add the related contributing documents and such as well as labels, etc..
We are planning to release the opensearch-helm standalone repo sometime either next week or the week after that.
Would definitely need community help to migrate it together.

Thanks.

**Comment by mprimeaux** [comment]
@peterzhuamazon One thing to note is that I had to heavily modify the chart and related values files for lint to pass. I'll ensure all of those changes are encapsulated into the PR.

**Comment by TheAlgo** [comment]
> @peterzhuamazon One thing to note is that I had to heavily modify the chart and related values files for lint to pass. I'll ensure all of those changes are encapsulated into the PR.

@mprimeaux what exactly do you mean by heavy modification of the chart. As far as I remember all the lint checks were passing last time I raised the PR. Curious to know what changes we need for publishing this 😀

**Comment by mprimeaux** [comment]
@TheAlgo Regarding lint rules, there were broadly four categories of lint errors:

* Too many blank lines 
* Trailing spaces
* Comments without a trailing space. 
* Missing CR/LF at the end of the yaml

You can look at the differences between current charts and relate values files for the detail. 

I’m happy to walk you through the differences as needed.

**Comment by TheAlgo** [comment]
@mprimeaux I get that. It might be possible as I did not use workflow actions to lint but the `helm lint` command which is used to find lint errors which can cause installation issues. 

In that case I agree it might be tedious to fix those. Thanks for taking us up. Much appreciated.

**Comment by mprimeaux** [comment]
@TheAlgo Thanks for replying and for that detail. The subtle difference is that `helm lint` checks for [helm-related errors](https://github.com/helm/helm/blob/efe2638f87b597403f34009e1029a6f0b44db8f2/pkg/action/lint.go#L49) (i.e missing chart version, etc) whereas the helm-chart-testing action checks for yaml errors.  

To your point, in my opinion, it would be a benefit to add yaml listing to helm.

**Comment by mprimeaux** [comment]
@peterzhuamazon Just checking in to see if you all are still on track for this week or next for the new repository. In the interest of transparency, we are keen to get this in place for nearly 7 multi-tenant 'as-a-Service' platforms globally we have at Cubic Transportation Systems in AWS.

Your help is greatly appreciated. We stand by ready to help.

**Comment by peterzhuamazon** [comment]
Hi @mprimeaux sorry for the late response we are on track to release the repo, probably a bit longer than expected since we are currently voting for the name of it here #49.

There were some initial concerns on the naming so we would like the community to vote on it before proceed.

Thanks.

**Comment by mprimeaux** [comment]
@peterzhuamazon Thanks for the update. 

I've cast [my vote](https://github.com/opensearch-project/opensearch-devops/issues/49#issuecomment-901497542) 👍🏼 in [#49](https://github.com/opensearch-project/opensearch-devops/issues/49).

**Comment by mprimeaux** [comment]
@peterzhuamazon Our team also has made defect fixes related to security that we will issue as a PR. Perhaps best in the new repository.

**Comment by peterzhuamazon** [comment]
> @peterzhuamazon Our team also has made defect fixes related to security that we will issue as a PR. Perhaps best in the new repository.

Thanks @mprimeaux we plan to start the migration next week, since the vote will end on Friday.
We will make the repo public next week as well once the name is settled.

You will then be able to contribute.

Thanks.
